metricbeat/module/couchdb/server/server.go

```diff
@@ -61,23 +71,108 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	return &MetricSet{
-		base,
-		http,
+		BaseMetricSet: base,
+		http:          http,
+		fetcher:       nil,
 	}, nil
 }
 
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
 func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
+	info, err := m.getInfoFromCouchdbHost(m.Host())
+	if err != nil {
+		return errors.Wrap(err, "cannot start couchdb metricbeat module")
+	}
+
+	v, err := getVersionFromInfo(info)
+	if err != nil {
+		return errors.Wrap(err, "error trying to parse version info")
+	}
+
+	switch v.major {
+	case "1":
+		m.fetcher = &V1{}
```

Comment:
Do you think this can be optimized to not create a new instance every time the `Fetch` is called?

Reply:
See above